ingestion/src/metadata/ingestion/source/deltalake.py

```diff
@@ -152,7 +197,59 @@ def _fetch_view_schema(self, view_name: str) -> Optional[Dict]:
                 view_detail[row_dict["col_name"]] = row_dict["data_type"]
             if "# Detailed Table" in row_dict["col_name"]:
                 col_details = True
-        return view_detail
+        return view_detail.get("View Text")
+
+    def _check_col_length(self, datatype, col_raw_type):
+        if datatype and datatype.upper() in {"CHAR", "VARCHAR", "BINARY", "VARBINARY"}:
+            try:
+                return col_raw_type.length if col_raw_type.length else 1
+            except AttributeError:
+                return 1
+
+    def _get_col_info(self, row):
+        parsed_string = ColumnTypeParser._parse_datatype_string(row["data_type"])
+        column = None
+        if parsed_string:
+            parsed_string["dataLength"] = self._check_col_length(
+                parsed_string["dataType"], row["data_type"]
+            )
+            if row["data_type"] == "array":
+                array_data_type_display = (
+                    repr(row["data_type"])
+                    .replace("(", "<")
+                    .replace(")", ">")
+                    .replace("=", ":")
+                    .replace("<>", "")
+                    .lower()
+                )
+                parsed_string["dataTypeDisplay"] = f"{array_data_type_display}"
+                parsed_string[
+                    "arrayDataType"
+                ] = ColumnTypeParser._parse_primitive_datatype_string(
+                    array_data_type_display[6:-1]
+                )[
+                    "dataType"
+                ]
+            column = Column(name=row["col_name"], **parsed_string)
+        else:
+            col_type = re.search(r"^\w+", row["data_type"]).group(0)
+            charlen = re.search(r"\(([\d]+)\)", row["data_type"])
+            if charlen:
+                charlen = int(charlen.group(1))
+            if (
+                col_type.upper() in {"CHAR", "VARCHAR", "VARBINARY", "BINARY"}
+                and charlen is None
+            ):
+                charlen = 1
+
+            column = Column(
+                name=row["col_name"],
+                description=row["comment"] if row["comment"] else None,
```

Comment:
it is good that we are checking if `comment` is part of `row`. But if it is not, the validation would be useless as we will get a KeyError. Let's use `row.get("comment")`.